src/App.tsx

```diff
@@ -1,223 +1,157 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorNotification } from
+  './components/Auth/ErrorNotification/ErrorNotification';
+import { Footer } from './components/Auth/Footer/Footer';
+import { Header } from './components/Auth/Header/Header';
+import { TodoList } from './components/Auth/TodoList/TodoList';
+import { ErrorMessage } from './types/Error';
+import { FilterType } from './types/FilterBy';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [title, setTitle] = useState('');
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+  const user = useContext(AuthContext);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch (error) {
+        setErrorMessage(`${error}`);
+      }
+    };
+
+    if (!user) {
+      return;
+    }
+
+    getTodosFromServer(user.id);
+  }, []);
+
+  const getFilteredTodo = useMemo(() => todos.filter(todo => {
+    switch (filterType) {
+      case FilterType.Active:
+        return !todo.completed;
+
+      case FilterType.Completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  }), [todos, filterType]);
```

Comment:
You can create a helpers folder with the file getFilteredTodos: function that return filtered todos (here will be less code)